### PR TITLE
Add llamatune checks with quantized values

### LINKED ISSUE #813: Add quantized tunable + llamatune tests
_Originally posted by @bpkroth in https://github.com/microsoft/MLOS/pull/812#discussion_r1688700411_

### PR DESCRIPTION
**Pull request by bpkroth**
Follow on to #812 

Closes #813

### CODE REVIEW

File: mlos_core/mlos_core/tests/spaces/adapters/llamatune_test.py [hunk 1]
```diff
@@ -26,8 +33,12 @@ def construct_parameter_space(
 
     for idx in range(n_continuous_params):
         input_space.add(CS.UniformFloatHyperparameter(name=f"cont_{idx}", lower=0, upper=64))
+    for idx in range(n_quantized_continuous_params):
+        input_space.add(CS.UniformFloatHyperparameter(name=f"cont_{idx}", lower=0, upper=64, q=12.8))
     for idx in range(n_integer_params):
         input_space.add(CS.UniformIntegerHyperparameter(name=f"int_{idx}", lower=-1, upper=256))
+    for idx in range(n_quantized_integer_params):
+        input_space.add(CS.UniformIntegerHyperparameter(name=f"int_{idx}", lower=0, upper=256, q=16))
```
Comment (bpkroth) [inline mlos_core/mlos_core/tests/spaces/adapters/llamatune_test.py:41]: FIXME: This is no longer the correct method of implementing quantization with ConfigSpace >= 1.0

**Comment by bpkroth** [comment]
@motus do you want to take over this one while you're working on #835 ?

**Review by motus — APPROVED** [review]
The code looks good. Let's make CI pass and merge it in. I don't care in which order this PR goes in, before or after the quantization update.

**Comment by motus** [comment]
< @motus do you want to take over this one while you're working on https://github.com/microsoft/MLOS/pull/835 ?
sure! let me fix it now

File: mlos_bench/mlos_bench/optimizers/convert_configspace.py [hunk 2]
```diff
@@ -20,14 +20,15 @@
     Normal,
     Uniform,
 )
-from ConfigSpace.functional import quantize
 from ConfigSpace.hyperparameters import NumericalHyperparameter
 from ConfigSpace.types import NotSet
 
 from mlos_bench.tunables.tunable import Tunable, TunableValue
 from mlos_bench.tunables.tunable_groups import TunableGroups
 from mlos_bench.util import try_parse_val
```
Comment (motus) [inline mlos_bench/mlos_bench/optimizers/convert_configspace.py:29]:
```suggestion
```

File: mlos_bench/mlos_bench/tests/tunables/tunable_to_configspace_test.py [hunk 3]
```diff
@@ -17,14 +17,15 @@
 
 from mlos_bench.optimizers.convert_configspace import (
     TunableValueKind,
-    _monkey_patch_quantization,
     _tunable_to_configspace,
     special_param_names,
     tunable_groups_to_configspace,
 )
 from mlos_bench.tunables.tunable import Tunable
 from mlos_bench.tunables.tunable_groups import TunableGroups
```
Comment (motus) [inline mlos_bench/mlos_bench/tests/tunables/tunable_to_configspace_test.py:26]:
```suggestion
```

File: mlos_core/mlos_core/tests/spaces/adapters/llamatune_test.py [hunk 4]
```diff
@@ -12,12 +12,20 @@
 import pandas as pd
 import pytest
 
+from mlos_core.spaces.converters.util import monkey_patch_quantization
 from mlos_core.spaces.adapters import LlamaTuneAdapter
```
Comment (motus) [inline mlos_core/mlos_core/tests/spaces/adapters/llamatune_test.py:17]:
```suggestion
from mlos_core.spaces.adapters import LlamaTuneAdapter
from mlos_core.spaces.converters.util import monkey_patch_quantization
```

**Review by motus — APPROVED** [review]

**Comment by bpkroth** [comment]
@motus concerned this isn't working quite right yet since the result output isn't quantized as expected (e.g., the test values that were reverted in the PR you made against mine in https://github.com/bpkroth/MLOS/pull/59).

Let's start another PR to make sure that is working as expected and we get the quantization in the the `meta` field as we discussed.

**Pull request merged**